Date: Sep 24 17:23:50 2020
From: ddfaecc531497437fee22daab47c0b2a2bc53aa2 Sep 24 17:23:50 2020
Subject: [PATCH] Added secure location for WPENW/WebProcess

Changing minidump descriptor to encrypted folder

Source: COMCAST
License: BSD-2-Clause & BSD-3-Clause & LGPL-2.0 & LGPL-2.1 
Upstream-Status: Pending
Signed-off-by: Matta Nancharayya <Nancharayya_Matta@comcast.com>
---
Index: git/Source/WebKit/Shared/unix/BreakpadExceptionHandler.cpp
===================================================================
--- git.orig/Source/WebKit/Shared/unix/BreakpadExceptionHandler.cpp
+++ git/Source/WebKit/Shared/unix/BreakpadExceptionHandler.cpp
@@ -22,6 +22,7 @@
 #include "BreakpadExceptionHandler.h"
 #include <client/linux/handler/exception_handler.h>
 #include <signal.h>
+#include <stdio.h>
 
 namespace WebKit
 {
@@ -41,6 +42,12 @@ void installExceptionHandler()
   static google_breakpad::ExceptionHandler* excHandler = NULL;
   delete excHandler;
   const char* BREAKPAD_MINIDUMP_DIR = "/opt/minidumps";
+  FILE *fp;
+  if (( fp = fopen("/tmp/.SecureDumpEnable", "r")) != NULL)
+  {
+	 BREAKPAD_MINIDUMP_DIR = "/opt/secure/minidumps";
+	fclose(fp);
+  }
   excHandler = new google_breakpad::ExceptionHandler(google_breakpad::MinidumpDescriptor(BREAKPAD_MINIDUMP_DIR), NULL, breakpadCallback, NULL, true, -1);
 }
 }
